p2p/resolve_ref_test.go

```diff
@@ -2,66 +2,150 @@ package p2p
+			if len(connectedPeers) == numNodes {
+				close(qriPeerConnWaitCh)
+			}
+		}
+		return nil
+	}
+	bus.Subscribe(watchP2PQriEvents, event.ETP2PQriPeerConnected)
+
+	// create a new, disconnected node
+	testnode, err := p2ptest.NewNodeWithBus(ctx, factory, bus)
 	if err != nil {
 		t.Fatal(err)
 	}
-	ref := reporef.DatasetRef{Peername: p.Peername, Name: "bar", ProfileID: p.ID, Path: "/ipfs/QmXSGsgt8Bn8jepw7beXibYUfWSJVU2SzP3TpkioQVUrmM"}
-	if err := peers[4].Repo.PutRef(ref); err != nil {
-		t.Fatalf("error putting ref in repo: %s", err.Error())
-	}
-	if err := repo.CanonicalizeDatasetRef(peers[4].Repo, &reporef.DatasetRef{Peername: p.Peername, Name: "bar"}); err != nil {
-		t.Fatalf("peer must be able to resolve local ref. error: %s", err.Error())
-	}
+	node = testnode.(*QriNode)
+	// closing the discovery process prevents situations where another peer finds
+	// our node in the wild and attempts to connect to it while we are trying
+	// to connect to that node at the same time. This causes a dial failure.
+	node.Discovery.Close()
+	defer node.GoOffline()
+	t.Log("main test node id: ", node.host.ID())
 
-	expect := "tim/bar@/ipfs/QmXSGsgt8Bn8jepw7beXibYUfWSJVU2SzP3TpkioQVUrmM"
-
-	t.Logf("testing ResolveDatasetRef message with %d peers", len(peers))
-	var wg sync.WaitGroup
-	for i, p := range peers {
-		if i != 4 {
-			wg.Add(1)
-			go func(p *QriNode) {
-				defer wg.Done()
-				ref := reporef.DatasetRef{Peername: "tim", Name: "bar"}
-				if err := p.ResolveDatasetRef(ctx, &ref); err != nil {
-					t.Errorf("%s ResolveDatasetRef error: %s", p.ID, err.Error())
-				}
-				if ref.String() != expect {
-					pro, _ := p.Repo.Profile()
-					t.Errorf("%s %s name mismatch: %s != %s", pro.Peername, p.ID, ref.String(), expect)
-					return
-				}
-			}(p)
+	// explicitly connect the main test node to each other node in the network
+	for _, groupNode := range nodes {
+		addrInfo := peer.AddrInfo{
+			ID:    groupNode.Host().ID(),
+			Addrs: groupNode.Host().Addrs(),
+		}
+		err := node.Host().Connect(context.Background(), addrInfo)
+		if err != nil {
+			t.Logf("error connecting to peer %q: %s", groupNode.Host().ID(), err)
 		}
 	}
 
-	wg.Wait()
+	<-qriPeerConnWaitCh
+
+	p2pRefResolver := node.NewP2PRefResolver()
+
+	dsrefspec.AssertResolverSpec(t, p2pRefResolver, func(r dsref.Ref, author identity.Author, _ *oplog.Log) error {
+		builder := dscache.NewBuilder()
+		pid, err := identity.KeyIDFromPub(author.AuthorPubKey())
+		builder.AddUser(r.Username, pid)
+		if err != nil {
+			return err
+		}
+		builder.AddDsVersionInfo(dsref.VersionInfo{
+			Username:  r.Username,
+			InitID:    r.InitID,
+			Path:      r.Path,
+			ProfileID: pid,
+			Name:      r.Name,
+		})
+		cache := builder.Build()
+		nodeWithRef.Repo.Dscache().Assign(cache)
+		testResolveRef := &dsref.Ref{
+			Username: r.Username,
+			Name:     r.Name,
+		}
+		nodeWithRef.Repo.Dscache().ResolveRef(ctx, testResolveRef)
+		t.Logf("test resolve ref %v\n", testResolveRef)
```

Comment:
I actually left it on purpose so you have more details about what happened, since the logging only show unless you specifiy -v or the test fails. But, happy to remove it!